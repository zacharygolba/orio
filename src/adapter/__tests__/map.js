// @flow

import Map from '../map'
import * as iter from '../../'

let subj

beforeEach(() => {
  const { producer } = iter.from([1, 2, 3])
  subj = new Map(producer, value => value * 2)
})

test('#@@iterator()', () => {
  for (const item of subj) {
    expect(item).toMatchSnapshot()
  }
})

test('#next()', () => {
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
})

test('#sizeHint()', () => {
  expect(subj.sizeHint()).toEqual(3)
})
