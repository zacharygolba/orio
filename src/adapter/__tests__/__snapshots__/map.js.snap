// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#@@iterator() 1`] = `2`;

exports[`#@@iterator() 2`] = `4`;

exports[`#@@iterator() 3`] = `6`;

exports[`#next() 1`] = `
Object {
  "done": false,
  "value": 2,
}
`;

exports[`#next() 2`] = `
Object {
  "done": false,
  "value": 4,
}
`;

exports[`#next() 3`] = `
Object {
  "done": false,
  "value": 6,
}
`;

exports[`#next() 4`] = `
Object {
  "done": true,
  "value": undefined,
}
`;
